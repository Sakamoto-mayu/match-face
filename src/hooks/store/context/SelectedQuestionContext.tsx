"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { selectedQuestionReducer } from "../reducer/selectedQuestionReducer";
import { SelectReducerAction } from "@/types/admin/tasks/register/types";

const SelectedQuestionContext = createContext<
  [string[], Dispatch<SelectReducerAction>] | undefined
>(undefined);

export const SelectedQuestionProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<string[], SelectReducerAction>>(
    selectedQuestionReducer,
    []
  );
  return (
    <SelectedQuestionContext.Provider value={[state, dispatch]}>
      {children}
    </SelectedQuestionContext.Provider>
  );
};

export const useSelectQuestion = () => {
  const context = useContext(SelectedQuestionContext);
  if (context === undefined)
    throw new Error("Function 'useUserSelect' must be used in Provider.");
  return context;
};
