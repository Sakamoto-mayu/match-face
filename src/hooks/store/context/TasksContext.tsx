"use client";

import React from "react";

// 職種フィルター用context
const JobsFilterContext = React.createContext<
  [string[], React.Dispatch<React.SetStateAction<string[]>>] | undefined
>(undefined);

export const JobsFilterProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const [filterlist, setFilterList] = React.useState<string[]>([]);
  return (
    <JobsFilterContext.Provider value={[filterlist, setFilterList]}>
      {children}
    </JobsFilterContext.Provider>
  );
};

export const useJobsFilter = () => {
  const context = React.useContext(JobsFilterContext);
  if (context === undefined) {
    throw new Error("useRefine must be used within a Provider");
  }
  return context;
};
// ここまで
