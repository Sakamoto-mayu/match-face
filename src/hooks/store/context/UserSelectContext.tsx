"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { userSelectReducer } from "../reducer/userSelectReducer";
import { UserSelectReducerAction } from "@/types/admin/tasks/register/types";

const UserSelectContext = createContext<
  [string[], Dispatch<UserSelectReducerAction>] | undefined
>(undefined);

export const CounterProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<string[], UserSelectReducerAction>>(
    userSelectReducer,
    []
  );
  return (
    <UserSelectContext.Provider value={[state, dispatch]}>
      {children}
    </UserSelectContext.Provider>
  );
};

export const useUserSelect = () => {
  const context = useContext(UserSelectContext);
  if (context === undefined)
    throw new Error("Function 'useCounter' must be used in Provider.");
  return context;
};
