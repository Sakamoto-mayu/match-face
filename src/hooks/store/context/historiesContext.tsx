'use client'
import { createContext, useContext } from "react";
import { Departments, Skills } from "@/types/admin/histories/admin_histories";
import { skills,departments } from "@/const/admin_histories";

export const SkillsContext = createContext<Skills|undefined>(undefined)
export const DepartmentsContext = createContext<Departments|undefined>(undefined)

// constからのimport(valueの値)をfetchで取得したデータに変える
// const skills = fetch('')
// const departments = fetch('')

export function HistoriesProvider({children}:{children:React.ReactElement}) {
    return (
        <SkillsContext.Provider value = {skills}>
        <DepartmentsContext.Provider value = {departments}>
            {children}
        </DepartmentsContext.Provider>
        </SkillsContext.Provider>
    )
}
export const useSkills = () => {
    const context = useContext(SkillsContext)
    if(context === undefined){
        throw new Error('skillsが見つかりません')
    }
    return context
}
export const useDepartments = () => {
    const context = useContext(DepartmentsContext)
    if(context === undefined){
        throw new Error('departmentsが見つかりません')
    }
    return context
}
