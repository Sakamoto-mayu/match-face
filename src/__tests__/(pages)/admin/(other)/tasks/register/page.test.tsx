import TaskRegisterPage from "@/app/(pages)/admin/(other)/tasks/register/page";
import { render } from "@testing-library/react";
import { UserSelectProvider } from "@/hooks/store/context/UserSelectContext";

// useContextをモーダルで使用しているため、Providerを追加しています

describe('タスク一覧画面', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    render(
      <UserSelectProvider>
        <TaskRegisterPage />
      </UserSelectProvider>
    );
  });
  afterAll(() => {
    jest.clearAllMocks();
  });

  it("レンダリング時", () => {
    const { container } = render(
      <UserSelectProvider>
        <TaskRegisterPage />
      </UserSelectProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
