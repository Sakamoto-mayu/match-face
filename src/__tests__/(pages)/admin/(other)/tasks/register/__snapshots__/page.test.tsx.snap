// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`タスク一覧画面 レンダリング時 1`] = `
<div>
  <main>
    <div
      class="border-2 border-light-gray w-3/5 mx-auto my-10 p-8"
    >
      <div>
        <div
          class="mb-5"
        >
          <div
            class="flex items-center mb-2"
          >
            <p
              class="text-base w-40"
            >
              ▶ユーザーを選択する
            </p>
            <div
              class="text-xs ml-10"
            >
              <button
                class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs"
              >
                追加
              </button>
            </div>
          </div>
          <div>
            <div>
              <input
                class="border-2 border-light-gray text-xs p-1 w-full"
                id="search"
                readonly=""
              />
            </div>
          </div>
        </div>
        <div
          class="mb-5"
        >
          <div
            class="flex items-center mb-2"
          >
            <p
              class="text-base w-40"
            >
              ▶質問を選択する
            </p>
            <div
              class="text-xs ml-10"
            >
              <button
                class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs"
              >
                追加
              </button>
            </div>
          </div>
          <div>
            <div>
              <input
                class="border-2 border-light-gray text-xs p-1 w-full"
                id="search"
                readonly=""
              />
            </div>
          </div>
        </div>
        <div>
          <div
            class="flex items-center"
          >
            <div>
              <p
                class="text-base w-40"
              >
                ▶締切日時を設定する
              </p>
            </div>
            <div
              class="flex items-center ml-10"
            >
              <div
                class="flex items-center"
              >
                <select
                  class="border-2 border-light-gray w-20 h-6 text-sm"
                  id="year"
                  name="year"
                >
                  <option
                    value="2023"
                  >
                    2023
                  </option>
                  <option
                    value="2024"
                  >
                    2024
                  </option>
                </select>
                <p
                  class="text-ms ml-0.5"
                >
                  年
                </p>
              </div>
              <div
                class="flex items-center ml-4"
              >
                <select
                  class="border-2 border-light-gray w-20 h-6 text-sm"
                  id="month"
                  name="month"
                >
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="6"
                  >
                    6
                  </option>
                  <option
                    value="7"
                  >
                    7
                  </option>
                  <option
                    value="8"
                  >
                    8
                  </option>
                  <option
                    value="9"
                  >
                    9
                  </option>
                  <option
                    value="10"
                  >
                    10
                  </option>
                  <option
                    value="11"
                  >
                    11
                  </option>
                  <option
                    value="12"
                  >
                    12
                  </option>
                </select>
                <p
                  class="text-ms ml-0.5"
                >
                  月
                </p>
              </div>
              <div
                class="flex items-center ml-4"
              >
                <select
                  class="border-2 border-light-gray w-20 h-6 text-sm"
                  id="day"
                  name="day"
                >
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="6"
                  >
                    6
                  </option>
                  <option
                    value="7"
                  >
                    7
                  </option>
                  <option
                    value="8"
                  >
                    8
                  </option>
                  <option
                    value="9"
                  >
                    9
                  </option>
                  <option
                    value="10"
                  >
                    10
                  </option>
                  <option
                    value="11"
                  >
                    11
                  </option>
                  <option
                    value="12"
                  >
                    12
                  </option>
                  <option
                    value="13"
                  >
                    13
                  </option>
                  <option
                    value="14"
                  >
                    14
                  </option>
                  <option
                    value="15"
                  >
                    15
                  </option>
                  <option
                    value="16"
                  >
                    16
                  </option>
                  <option
                    value="17"
                  >
                    17
                  </option>
                  <option
                    value="18"
                  >
                    18
                  </option>
                  <option
                    value="19"
                  >
                    19
                  </option>
                  <option
                    value="20"
                  >
                    20
                  </option>
                  <option
                    value="21"
                  >
                    21
                  </option>
                  <option
                    value="22"
                  >
                    22
                  </option>
                  <option
                    value="23"
                  >
                    23
                  </option>
                  <option
                    value="24"
                  >
                    24
                  </option>
                  <option
                    value="25"
                  >
                    25
                  </option>
                  <option
                    value="26"
                  >
                    26
                  </option>
                  <option
                    value="27"
                  >
                    27
                  </option>
                  <option
                    value="28"
                  >
                    28
                  </option>
                  <option
                    value="29"
                  >
                    29
                  </option>
                  <option
                    value="30"
                  >
                    30
                  </option>
                  <option
                    value="31"
                  >
                    31
                  </option>
                </select>
                <p
                  class="text-ms ml-0.5"
                >
                  日
                </p>
              </div>
              <div
                class="flex items-center ml-4"
              >
                <select
                  class="border-2 border-light-gray w-20 h-6 text-sm"
                  id="time"
                  name="time"
                >
                  <option
                    value="0"
                  >
                    0
                  </option>
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="6"
                  >
                    6
                  </option>
                  <option
                    value="7"
                  >
                    7
                  </option>
                  <option
                    value="8"
                  >
                    8
                  </option>
                  <option
                    value="9"
                  >
                    9
                  </option>
                  <option
                    value="10"
                  >
                    10
                  </option>
                  <option
                    value="11"
                  >
                    11
                  </option>
                  <option
                    value="12"
                  >
                    12
                  </option>
                  <option
                    value="13"
                  >
                    13
                  </option>
                  <option
                    value="14"
                  >
                    14
                  </option>
                  <option
                    value="15"
                  >
                    15
                  </option>
                  <option
                    value="16"
                  >
                    16
                  </option>
                  <option
                    value="17"
                  >
                    17
                  </option>
                  <option
                    value="18"
                  >
                    18
                  </option>
                  <option
                    value="19"
                  >
                    19
                  </option>
                  <option
                    value="20"
                  >
                    20
                  </option>
                  <option
                    value="21"
                  >
                    21
                  </option>
                  <option
                    value="22"
                  >
                    22
                  </option>
                  <option
                    value="23"
                  >
                    23
                  </option>
                  <option
                    value="24"
                  >
                    24
                  </option>
                </select>
                <p
                  class="text-ms ml-0.5"
                >
                  時
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex justify-center mt-10"
        >
          <div>
            <button
              class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-base"
            >
              タスクを追加する
            </button>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
